dbms/src/Flash/Mpp/MPPTunnel.cpp

```diff
@@ -214,13 +214,13 @@ void MPPTunnel::connectLocal(size_t source_index, LocalRequestHandler & local_re
         LOG_TRACE(log, "ready to connect local");
         if (is_fine_grained)
         {
-            local_tunnel_fine_grained_sender = std::make_shared<LocalTunnelSender<true>>(source_index, local_request_handler, log, mem_tracker, tunnel_id);
-            tunnel_sender = local_tunnel_fine_grained_sender;
+            local_tunnel_fine_grained_sender_v2 = std::make_shared<LocalTunnelSenderV2<true>>(source_index, local_request_handler, log, mem_tracker, tunnel_id);
+            tunnel_sender = local_tunnel_fine_grained_sender_v2;
         }
         else
         {
-            local_tunnel_sender = std::make_shared<LocalTunnelSender<false>>(source_index, local_request_handler, log, mem_tracker, tunnel_id);
-            tunnel_sender = local_tunnel_sender;
+            local_tunnel_sender_v2 = std::make_shared<LocalTunnelSenderV2<false>>(source_index, local_request_handler, log, mem_tracker, tunnel_id);
+            tunnel_sender = local_tunnel_sender_v2;
```

Comment:
It is available to use V2 directly?

Reply:
> It is available to use LocalTunnelSenderV2 directly?

sry, what does using LocalTunnelSenderV2 directly means?

Reply:
There is a `LocalTunnelSenderV1`. I just want to know why `LocalTunnelSenderV1` is not used.

Reply:
> There is a structure called `LocalTunnelSenderV1`. I just want to know why `LocalTunnelSenderV1` is not used.

No, we use it in MPPTunnel::connectLocalV1

Reply:
Do you mean `MPPTunnel::connectLocal` is only for V2?